content/blog/2023/2023-08-28-fairness-adversarial-debiasing.md

```diff
@@ -0,0 +1,59 @@
+
+{{< window-screenshot src="/blog_img/2023/2023-08-28-fairness-adversarial-debiasing.png" >}}
+
+As seen from the image, the Adversarial Debiasing widget's hyperparameters are as follows:
+
+- Neurons in hidden layers: The number of neurons in each of the hidden layers of the neural network.
+- Number of epochs: The number of epochs to train the neural network. This is the number of times the model will see the entire dataset.
+- Batch size: The number of instances to be used in each training batch. The model will be trained on batches of instances instead of the entire dataset simultaneously.
+- Use Debiasing: Whether to use the debiasing or not. The widget will function as a regular neural network model if this option is not selected.
+- Adversary loss weight: The weight of the adversary loss in the total loss function. The adversary loss is the loss function of the discriminator. The higher the weight, the more the model will try to reduce the discriminator's ability to predict the protected variable at the expense of the predictor's ability to predict the target variable.
+
+
+## Orange use case
+
+Now that we know how the Adversarial Debiasing widget works and how to use it, let us look at a real-world example for a classification task. For this example, we will use the [Adult dataset](https://archive.ics.uci.edu/ml/datasets/adult), which we have used before. Unlike before, we will not use the As Fairness widget to select fairness attributes; instead, we will keep the default ones, "sex" for the protected attribute and "male" for the privileged protected attribute value. In this example, we will train two Adversarial Debiasing models, one with and one without debiasing. We will also train a Random Forest model for comparison. We will then compare the results of the three models.
+
+{{< window-screenshot src="/blog_img/2023/2023-08-28-fairness-adversarial-debiasing-use-case.png" >}}
+
+{{< window-screenshot src="/blog_img/2023/2023-08-28-fairness-adversarial-debiasing-scores.png" >}}
+
+The Adversarial Debiasing model results show that fairness metrics, particularly Disparate Impact and Statistical Parity Difference, have improved when debiasing. When not using debiasing, the results are very similar to that of the Random Forest model. When using debiasing, their values moved closer to a fair baseline, signifying reduced biases for these metrics. The Disparate Impact moved from 0.294 to 1.051, suggesting that the model's predictions became more balanced regarding favorable outcomes for privileged and unprivileged groups. The Statistical Parity Difference went from -0.180 to 0.006, indicating a near-zero bias between these groups regarding favorable outcomes.
```

Comment:
The Adversarial Debiasing model results show that fairness metrics, particularly Disparate Impact and Statistical Parity Difference, have improved when debiasing. -> Test and Score shows that debiasing improved fairness metrics, particularly Disparate Impact and Statistical Parity Difference.

When not using debiasing -> Without debiasing

very similar -> similar

When using debiasing, their values moved closer to a fair baseline, signifying reduced biases for these metrics. -> delete, you are just repeating yourself

Continue with "Debiasing improved Disparate Impact from 0.294 to 1.051." And delete "suggesting that...." because we see the improvement and do not need yet another tedious definition.